From 898703c50f7e1d1661a6e012af1236c179abb46a Mon Sep 17 00:00:00 2001
From: Hien Huynh <hien.huynh.px@rvc.renesas.com>
Date: Mon, 9 Apr 2018 16:28:42 +0700
Subject: [PATCH 488/628] ARM: DTS: r8a7742: move mmc node right after dma node

issue "sh_mmcif ee220000.mmc: could not find pctldev for node /dma-controller@e6700000, deferring probe"
appeared on some iwg21m board, the issue was not investigated deeply, but move mmc node right after dma
node can prevent the issue

Signed-off-by: Hien Huynh <hien.huynh.px@rvc.renesas.com>
Signed-off-by: vietn <vietn@fsoft.com.vn>
---
 arch/arm/boot/dts/r8a7742.dtsi | 52 +++++++++++++++++++++---------------------
 1 file changed, 26 insertions(+), 26 deletions(-)

diff --git a/arch/arm/boot/dts/r8a7742.dtsi b/arch/arm/boot/dts/r8a7742.dtsi
index 186e28a..f5f038d 100644
--- a/arch/arm/boot/dts/r8a7742.dtsi
+++ b/arch/arm/boot/dts/r8a7742.dtsi
@@ -305,6 +305,32 @@
 		dma-channels = <15>;
 	};
 
+	mmcif0: mmc@ee200000 {
+		compatible = "renesas,mmcif-r8a7742", "renesas,sh-mmcif";
+		reg = <0 0xee200000 0 0x80>;
+		interrupts = <0 169 IRQ_TYPE_LEVEL_HIGH>;
+		clocks = <&mstp3_clks R8A7742_CLK_MMCIF0>;
+		dmas = <&dmac0 0xd1>, <&dmac0 0xd2>;
+		dma-names = "tx", "rx";
+		power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
+		reg-io-width = <4>;
+		status = "disabled";
+		max-frequency = <97500000>;
+	};
+
+	mmcif1: mmc@ee220000 {
+		compatible = "renesas,mmcif-r8a7742", "renesas,sh-mmcif";
+		reg = <0 0xee220000 0 0x80>;
+		interrupts = <0 170 IRQ_TYPE_LEVEL_HIGH>;
+		clocks = <&mstp3_clks R8A7742_CLK_MMCIF1>;
+		dmas = <&dmac0 0xe1>, <&dmac0 0xe2>;
+		dma-names = "tx", "rx";
+		power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
+		reg-io-width = <4>;
+		status = "disabled";
+		max-frequency = <97500000>;
+	};
+
 	audma0: dma-controller@ec700000 {
 		compatible = "renesas,rcar-dmac";
 		reg = <0 0xec700000 0 0x10000>;
@@ -363,32 +389,6 @@
 		dma-channels = <13>;
 	};
 
-	mmcif0: mmc@ee200000 {
-		compatible = "renesas,mmcif-r8a7742", "renesas,sh-mmcif";
-		reg = <0 0xee200000 0 0x80>;
-		interrupts = <0 169 IRQ_TYPE_LEVEL_HIGH>;
-		clocks = <&mstp3_clks R8A7742_CLK_MMCIF0>;
-		dmas = <&dmac0 0xd1>, <&dmac0 0xd2>;
-		dma-names = "tx", "rx";
-		power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
-		reg-io-width = <4>;
-		status = "disabled";
-		max-frequency = <97500000>;
-	};
-
-	mmcif1: mmc@ee220000 {
-		compatible = "renesas,mmcif-r8a7742", "renesas,sh-mmcif";
-		reg = <0 0xee220000 0 0x80>;
-		interrupts = <0 170 IRQ_TYPE_LEVEL_HIGH>;
-		clocks = <&mstp3_clks R8A7742_CLK_MMCIF1>;
-		dmas = <&dmac0 0xe1>, <&dmac0 0xe2>;
-		dma-names = "tx", "rx";
-		power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
-		reg-io-width = <4>;
-		status = "disabled";
-		max-frequency = <97500000>;
-	};
-
 	pfc: pfc@e6060000 {
 		compatible = "renesas,pfc-r8a7742";
 		reg = <0 0xe6060000 0 0x250>;
-- 
2.7.4

